extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfigUtil.java

```diff
@@ -96,6 +97,12 @@ private static void handleExpose(Config config, Map<String, Object> unPrefixed)
                     .orElse(false);
             if (unprefixedExpose || prefixedExpose) {
                 unPrefixed.put(DEKORATE_PREFIX + generator + "." + EXPOSE_PROPERTY_NAME, true);
+                for (PlatformConfiguration platformConfiguration : platformConfigurations) {
+                    if (platformConfiguration.getConfigName().equals(generator)) {
+                        platformConfiguration.getHost().map(h -> unPrefixed.put(DEKORATE_PREFIX + generator + ".host", h));
```

Comment:
```suggestion
                        platformConfiguration.getHost().ifPresent(h -> unPrefixed.put(DEKORATE_PREFIX + generator + ".host", h));
```

Reply:
Or the whole for/if/break block could be written using streams :)

Reply:
```java
Arrays.stream(platformConfigurations)
    .filter(f->f.getConfigName().equals(generator))
    .map(PlatformConfiguration::getHost)
    .findFirst()
    .ifPresent(h-> unPrefixed.put(DEKORATE_PREFIX + generator + ".host", h))
```

Reply:
Yes to first, no to the second :)